pkg/controller/customresourcedefinition/customresourcedefinition_controller.go

```diff
@@ -0,0 +1,238 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package customresourcedefinition
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/awslabs/aws-eks-cluster-controller/pkg/finalizers"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/awslabs/aws-eks-cluster-controller/pkg/authorizer"
+	"github.com/awslabs/aws-eks-cluster-controller/pkg/logging"
+	"go.uber.org/zap"
+
+	clusterv1alpha1 "github.com/awslabs/aws-eks-cluster-controller/pkg/apis/cluster/v1alpha1"
+	componentsv1alpha1 "github.com/awslabs/aws-eks-cluster-controller/pkg/apis/components/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	apiextv1beta "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// statuses and finalizers
+var (
+	CRDFinalizer        = "crd.components.eks.amazon.com"
+	RemoteObjectCreated = "Created"
+)
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new CustomResourceDefinition Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	sess := session.Must(session.NewSession())
```

Comment:
MINOR: I think this might need to be changed in every controller -  but we should set the AWS_REGION for this - maybe with an env var or something.

Reply:
Gave this some thought, I am not sure if we want to set a default region for this. I am wary this will lead to some unexpected surprises for the user. Let me know if you feel otherwise or if i missed something. Will change